stdlib/statistics.pyi

```diff
@@ -18,7 +19,13 @@ if sys.version_info >= (3, 8):
     def geometric_mean(data: Iterable[SupportsFloat]) -> float: ...
 
 def mean(data: Iterable[_NumberT]) -> _NumberT: ...
-def harmonic_mean(data: Iterable[_NumberT]) -> _NumberT: ...
+
+if sys.version_info >= (3, 10):
+    def harmonic_mean(data: Iterable[_NumberT], weights: Optional[Iterable[_Number]] = ...) -> _NumberT: ...
```

Comment:
I think this should also be `_NumberT`; `statistics.harmonic_mean([1, 2.0], weights=[Decimal(3), Decimal(4)])` throws.